src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -463,22 +447,9 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 												   colocateWithTableName,
 												   viaDeprecatedAPI);
 
-	/*
-	 * Due to dropping columns, the parent's distribution key may not match the
-	 * partition's distribution key. The input distributionColumn belongs to
-	 * the parent. That's why we override the distribution column of partitions
-	 * here. See issue #5123 for details.
-	 */
-	if (PartitionTable(relationId))
-	{
-		Oid parentRelationId = PartitionParentOid(relationId);
-		char *distributionColumnName =
-			ColumnToColumnName(parentRelationId, nodeToString(distributionColumn));
-
-		distributionColumn =
-			FindColumnWithNameOnTargetRelation(parentRelationId, distributionColumnName,
-											   relationId);
-	}
+	Var *distributionColumn = BuildDistributionKeyFromColumnName(relationId,
```

Comment:
The lock now comes in a little bit late. It's not necessarily problematic, but maybe to minimize the behavioural change, we use LockRelationOid at the start of this function?